### PR TITLE
Refactoring python project

### PR DESCRIPTION
**Pull request by gurevichdmitry**
### Summary of your changes

This pull request simplifies the codebase by combining two separate projects into a single one.

- Moved the `fleet_api` package from the `test-environments` to the `tests` folder, reorganizing it into `fleet_api` (common library) and `integrations_setup` (scripts and management).
- Updated dependencies accordingly.
- Adjusted telemetry tests to utilize the new `fleet_api` package location.
- Enhanced the `commonlib` to leverage the `fleet_api` package.
- Enabled the `k8s-ci` workflow on pull requests to validate changes.
- Updated workflows to support latest changes.
- Fixed `cloudformation ci` workflow: [successful run](https://github.com/elastic/cloudbeat/actions/runs/7553754163/job/20565179457?pr=1813)

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

File: .github/workflows/k8s-ci.yml [hunk 1]
```diff
@@ -3,11 +3,11 @@ name: K8s-CI
...
-  # pull_request:
-  #   branches:
-  #     - main
-  #     - "[0-9]+.[0-9]+"
-  #   types: [ opened, synchronize, reopened ]
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+"
+    types: [ opened, synchronize, reopened ]
```
Comment (gurevichdmitry) [inline .github/workflows/k8s-ci.yml:10]: @oren-zohar, after enabling `k8s-ci` on pre-merge, it appears to be stable and functioning well. Are we planning to continue with `pre-merge` tests, or is there consideration to enable the full version of tests only on post-merge? Additionally, we've rerun flaky tests workflow, which has contributed to improved stability.

File: tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml
Comment (oren-zohar) [inline tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml:1]: We mentioned changing the `deploy` folder name to something more informative, right?

**Review by oren-zohar — COMMENTED** [review]
Do we need to update the tests/tests/Readme.md?

File: tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml
Comment (gurevichdmitry) [reply tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml:1]: Yes, you are correct. I will proceed to change the deploy folder name to make it more informative.

**Comment by gurevichdmitry** [comment]
> Do we need to update the tests/tests/Readme.md?

I'll update [this](https://github.com/elastic/cloudbeat/blob/c507a2b88b8abaca48f4ca50b5dcf28cf7cc4545/tests/integrations_setup/README.md) readme

File: tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml
Comment (gurevichdmitry) [reply tests/deploy/k8s-cloudbeat-tests/templates/tests/cloudbeat-test-pod.yml:1]: `deploy` folder name was replaced by `test_environments`

**Review by oren-zohar — APPROVED** [review]

**Comment by oren-zohar** [comment]
@gurevichdmitry should this handle the issue with the Cloud Formation test?

**Comment by gurevichdmitry** [comment]
> * successful run

Sure, as evidence, I've provided a link to the successful run of the [CloudFormation CI](https://github.com/elastic/cloudbeat/actions/runs/7553754163/job/20565179457?pr=1813)

**Comment by gurevichdmitry** [comment]
> @gurevichdmitry should this handle the issue with the Cloud Formation test?

Sure, as evidence, I've provided a link to the successful run of the [CloudFormation CI](https://github.com/elastic/cloudbeat/actions/runs/7553754163/job/20565179457?pr=1813)

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**